test/queue_test.rb

```diff
@@ -129,6 +128,30 @@ def enqueue(method, *args)
     QC.default_queue = nil
   end
 
+  def test_multi_threaded_server
+    adapter1 = QC::ConnAdapter.new
+    adapter2 = QC::ConnAdapter.new
+    q1 = q2 = nil
+
+    QC.default_conn_adapter = adapter1
+
+    t1 = Thread.new do
+      QC.default_conn_adapter = adapter1
+      q1 = QC::Queue.new('queue1').conn_adapter
+    end
+
+    t2 = Thread.new do
+      QC.default_conn_adapter = adapter2
+      q2 = QC::Queue.new('queue2').conn_adapter
+    end
+
+    t1.join
+    t2.join
+
+    assert_equal adapter1, q1
+    assert_equal adapter2, q2
+  end
```

Comment:
@smathieu let's include a second test case, which basically does the same thing but without assigning a `QC::ConnAdapter`. This should make sure that different threads acquire their own connection when nothing else is specified.